### ISSUE TITLE
Environment variables starting with `BEGIN_` should not pass validation

### ISSUE THREAD

**Issue by macdonst** [beginner-corp/cli#144]
The CLI/API should not allow users to set environment variables that start with `BEGIN_`.

**Comment by ryanblock** [comment]
This should be an API side validation error!

**Comment by macdonst** [comment]
@ryanblock yup, I realize the fix will lie in the API code but I wanted to make sure we test it from the CLI to return a good error message.
